src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -251,6 +252,9 @@ private void configureSession(String user) throws IOException, GeneralSecurityEx
     configureHostChecking();
     if (this.preferredAuthenticationMethods != null && !this.preferredAuthenticationMethods.isEmpty()) {
       CoreModuleProperties.PREFERRED_AUTHS.set(client, this.preferredAuthenticationMethods.toLowerCase());
+    } else {
+      CoreModuleProperties.PREFERRED_AUTHS.set(client, SftpAuthenticationMethod.PUBLIC_KEY + "," +
```

Comment:
Can you puta log here to check under which condition and how many such call come here. This will be helpful for issue debugging also in future.

Reply:
Added debug logger.

Reply:
done